### PR TITLE
Push wildfly docker image on master build

### PR DESCRIPTION
**Pull request by volkflo**
I would like to push the docker image of wildfly on each master build.
I am not sure if this already works because I am not very familiar with github actions.

@EricWittmann Can you please take a look at it? :)

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -46,3 +46,9 @@ jobs:
         run: mvn test install -Dinvoker.skip -Dmaven.javadoc.skip=true -B -Dapiman.gateway-test.config=vertx3-file
       - name: Run Tests (AMG 1)
         run: mvn test install -Dinvoker.skip -Dmaven.javadoc.skip=true -B -Dapiman.gateway-test.config=amg-1
+      - name: Build and Publish Docker Images
+        if: github.ref == 'refs/heads/master'
+        run: |
+          mvn clean package docker:build -P docker -DskipTests
```
Comment (EricWittmann) [inline .github/workflows/verify.yaml:52]: Have you tried this locally?  Does it work?  I didn't think we had support for building a docker image using maven.  But looking at the code I *was* able to find a dockerfile and some configuration in the pom.xml.  So I guess this works?  If you can confirm that this produces what we want, then I can add the secrets and merge this PR.  It looks good.

**Comment by msavy** [comment]
This was something I contributed. It uses/used the Docker Maven Plugin to
build a master Docker image.

>
>

**Comment by msavy** [comment]
See https://github.com/apiman/apiman/blob/master/distro/wildfly/src/main/docker/Dockerfile
And https://github.com/apiman/apiman/blob/master/distro/wildfly/pom.xml#L145-L198

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (volkflo) [reply .github/workflows/verify.yaml:52]: Yes, this should work after the project build has passed. But I am not sure if the `github.ref == 'refs/heads/master'` part is correct

**Comment by volkflo** [comment]
> This was something I contributed. It uses/used the Docker Maven Plugin to build a master Docker image.
> […](#)

I changed this to `latest` because this more common I think.

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (EricWittmann) [reply .github/workflows/verify.yaml:52]: Yeah that should be fine - it just limits the step to the `master` branch.  Which I think is fine for us.

**Comment by EricWittmann** [comment]
Yeah the change to `latest` is correct.

**Comment by EricWittmann** [comment]
I'm happy with this if you want to mark is ready for review.

**Comment by volkflo** [comment]
> I'm happy with this if you want to mark is ready for review.

Ready for review :)

**Comment by msavy** [comment]
IMO, you should then publish to both names (or alias in some way),
otherwise people using master will end up with stale images and it won't
behave consistently.

**Comment by volkflo** [comment]
@msavy For the new image (apiman/on-wildfly (without a version)) there is no master at the moment. 
See: https://hub.docker.com/r/apiman/on-wildfly/tags

**Comment by msavy** [comment]
Okay, cool, thanks for that info. It would be good to add that back at some point.

**Comment by volkflo** [comment]
@EricWittmann Can we try this? :)

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**